### PR TITLE
Error handling and validation (discount & category)

### PR DESCRIPTION
**Pull request by VolodymyrBushko**

### CODE REVIEW

File: src/main/java/com/exadel/discountwebapp/category/service/CategoryService.java [hunk 1]
```diff
@@ -41,12 +43,10 @@ public CategoryResponseVO create(CategoryRequestVO request) {
...
-        Category category = categoryRepository.findById(id).orElse(null);
-        if (category != null) {
-            categoryMapper.updateEntity(request, category);
-            return categoryMapper.toVO(categoryRepository.save(category));
-        }
-        return null;
+        Category category = categoryRepository.findById(id).
+                orElseThrow(() -> new EntityNotFoundException("Could not find category with id: " + id));
+        categoryMapper.updateEntity(request, category);
+        return categoryMapper.toVO(categoryRepository.save(category));
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/category/service/CategoryService.java]: it is better to perform separate operations in different lines

categoryRepository.save(category)
categoryMapper.toVO(...

File: src/main/java/com/exadel/discountwebapp/category/vo/CategoryBaseVO.java [hunk 2]
```diff
@@ -3,13 +3,21 @@
...
 @SuperBuilder
 @NoArgsConstructor
 public abstract class CategoryBaseVO implements Serializable {
 
+    @NotBlank
+    @Size(min = 2, max = 25)
     private String title;
+
+    @Size(max = 510)
+    @Pattern(regexp = "^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]")
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/category/vo/CategoryBaseVO.java]: for now we can ommit this validation
it is too dificult. 
Did you write this expression yourself?

File: src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java [hunk 3]
```diff
@@ -41,12 +43,10 @@ public DiscountResponseVO create(DiscountRequestVO request) {
...
-        Discount discount = discountRepository.findById(id).orElse(null);
-        if (discount != null) {
-            discountMapper.updateEntity(request, discount);
-            return discountMapper.toVO(discountRepository.save(discount));
-        }
-        return null;
+        Discount discount = discountRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Could not find discount with id: " + id));
+        discountMapper.updateEntity(request, discount);
+        return discountMapper.toVO(discountRepository.save(discount));
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java]: the same here
please split this into two lines

Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java]: done

File: src/main/java/com/exadel/discountwebapp/category/service/CategoryService.java (hunk 1 above)
Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/category/service/CategoryService.java]: done

File: src/main/java/com/exadel/discountwebapp/category/vo/CategoryBaseVO.java (hunk 2 above)
Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/category/vo/CategoryBaseVO.java]: The regexp from StackOverflow. Removed

**Review by dmitriychernov7 — APPROVED** [review]

**Pull request merged**